### PR TITLE
added Link to categories

### PR DESCRIPTION
**Pull request by tusharkandpal**
- added Link to categories to re-direct to specific category products
- added 404PageNotFound page
- refactored code

### CODE REVIEW

File: src/components/sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
   const [sidebarToggle, setSidebarToggle] = useState("");
   const { productState, productDispatch } = useProduct();
   const { sortByPrice, categories, priceRange, rating } = productState.filters;
+  const [params] = useSearchParams();
+  const categoryParam = params.get("categoryName");
+
+  useEffect(() => {
+    if (categoryParam)
+      productDispatch({
+        type: "CATEGORY",
```
Comment (rehanmohiuddin) [inline src/components/sidebar/Sidebar.jsx:16]: define type as contants keep in seperate files and import

**Review by rehanmohiuddin — APPROVED** [review]

File: src/components/sidebar/Sidebar.jsx (hunk 1 above)
Comment (tusharkandpal) [reply src/components/sidebar/Sidebar.jsx:16]: will do!

File: src/components/sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+import { useState, useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
 import { useProduct } from "../../context/context";
 
 export function Sidebar() {
   const [sidebarToggle, setSidebarToggle] = useState("");
   const { productState, productDispatch } = useProduct();
   const { sortByPrice, categories, priceRange, rating } = productState.filters;
+  const [params] = useSearchParams();
+  const categoryParam = params.get("categoryName");
```
Comment (JantuDeb) [inline src/components/sidebar/Sidebar.jsx]: You might name it as `categoryName`

**Review by JantuDeb — APPROVED** [review]
Overall the code looks good

File: src/components/sidebar/Sidebar.jsx (hunk 2 above)
Comment (tusharkandpal) [reply src/components/sidebar/Sidebar.jsx]: sure will do.

**Pull request merged**